### PR TITLE
[stable/weave-scope] Fixes compatibility with 1.16

### PR DESCRIPTION
**Pull request by daixiang0**
To support 1.16. A quick overview of deprecated APIs can be found on their blog post: https://kubernetes.io/blog/2019/07/18/api-deprecations-in-1-16/.

Signed-off-by: Xiang Dai <764524258@qq.com>

### CODE REVIEW

File: stable/weave-scope/Chart.yaml
Comment (fbarl) [inline stable/weave-scope/Chart.yaml:3]: It would be nice if the `version` patch number followed `appVersion`'s. Luckily, Scope got a [new release](https://github.com/weaveworks/scope/releases) in the meantime, so maybe the easiest thing to do would be to bump the `appVersion` to `1.11.6` in this PR :)

File: stable/weave-scope/charts/weave-scope-agent/templates/_helpers.tpl
Comment (fbarl) [inline stable/weave-scope/charts/weave-scope-agent/templates/_helpers.tpl]: How about this?

```suggestion
{{- else -}}
```

In other words, use `apps/v1` as a _default_ and only regress to `extensions/v1beta1` if K8s version is old.

Comment (fbarl) [reply stable/weave-scope/charts/weave-scope-agent/templates/_helpers.tpl]: Also, I'd check for `1.9` instead of `1.14`, the docs you provided say that `apps/v1` is available since v1.9.

File: stable/weave-scope/charts/weave-scope-cluster-agent/templates/_helpers.tpl
Comment (fbarl) [inline stable/weave-scope/charts/weave-scope-cluster-agent/templates/_helpers.tpl:72]: Please see the comments above

File: stable/weave-scope/Chart.yaml
Comment (fbarl) [inline stable/weave-scope/Chart.yaml:5]: Idea (not for this PR): we might want to start specifying `kubeVersion` at some point, especially when we drop support for old K8s.

**Review by fbarl — CHANGES_REQUESTED** [review]
Thanks for your PR @daixiang0, good catch switching to the new `apiVersion`! :+1: 

I left a few comments, all should be easy to address.

File: stable/weave-scope/Chart.yaml
Comment (daixiang0) [reply stable/weave-scope/Chart.yaml:5]: I would like to do it when ensuure the minimum version number.

**Comment by daixiang0** [comment]
@fbarl test pass!

**Comment by fbarl** [comment]
Hey @daixiang0, thanks for addressing my comments and sorry for the late reply!

I see you've updated the `version` to `1.11.6`, but what I really meant is that you can bump the `appVersion` instead so that we'd have:

```
version: 1.1.6
appVersion: 1.11.6
```

Could you still do that please?

Thinking about it, it might be a hassle to maintain the patch version match in the future, but it would be good bump to the new release of Scope in this PR while you're at it :)

**Comment by daixiang0** [comment]
@fbarl done.

**Comment by daixiang0** [comment]
@fbarl any news?

**Comment by zanhsieh** [comment]
@fbarl
@dholbach
@bboreham
Would you help to LGTM this PR please? Thank you.

File: stable/weave-scope/Chart.yaml
Comment (hiddeco) [reply stable/weave-scope/Chart.yaml:3]: Given that chart changes can happen while the application version stays the same (e.g. due to breaking changes in Kubernetes), it is not likely this will be aligned for a long time.

But given

> Thinking about it, it might be a hassle to maintain the patch version match in the future, but it would be good bump to the new release of Scope in this PR while you're at it :)

you probably reached the same conclusion.

**Review by hiddeco — APPROVED** [review]
LGTM

File: stable/weave-scope/charts/weave-scope-agent/Chart.yaml
Comment (fbarl) [inline stable/weave-scope/charts/weave-scope-agent/Chart.yaml]: Could you please make this the same as in `weave-scope/Chart.yaml` above, just to make it consistent for now?

```
version: 1.1.6
appVersion: 1.11.6
```

File: stable/weave-scope/charts/weave-scope-cluster-agent/Chart.yaml
Comment (fbarl) [inline stable/weave-scope/charts/weave-scope-cluster-agent/Chart.yaml]: Same here, could you please change this to?

```
version: 1.1.6
appVersion: 1.11.6
```

**Review by fbarl — CHANGES_REQUESTED** [review]
Thanks for baring with me on this one @daixiang0, but there are now a lot of inconsistencies between `version` and `appVersion` numbers. Could you please address my two comments before we merge this?

p.s. I'll try to be more responsive next time.

**Review by fbarl — CHANGES_REQUESTED** [review]
I'd not merge before at least the `version` numbers are brought in order.

**Comment by dholbach** [comment]
Thanks @fbarl - sorry I missed this.

/hold

**Comment by daixiang0** [comment]
@dholbach @fbarl done

**Review by fbarl — APPROVED** [review]
Thanks for the change, LGTM! :tada:

**Comment by avarf** [comment]
Are you sure the changes fixes the compability problems with k8s v1.16?
I wanted to install it on k8s v1.16.3 and I got below error:
```
helm install stable/weave-scope --name weave-scope --namespace=logging

Error: validation failed: [unable to recognize "": no matches for kind "DaemonSet" in version "extensions/v1beta1", unable to recognize "": no matches for kind "Deployment" in version "extensions/v1beta1", unable to recognize "": no matches for kind "Deployment" in version "apps/v1beta1"]
```

**Comment by fbarl** [comment]
Good catch capturing this @avarf!

I managed to reproduce your error and I made a fix for it in #19400.

**Pull request merged**
